lib/app/services/database/database_manager_service.r.dart

```diff
@@ -0,0 +1,156 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'package:drift/drift.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:ion/app/features/chat/model/database/chat_database.m.dart';
+import 'package:ion/app/features/feed/data/database/following_feed_database/following_feed_database.m.dart';
+import 'package:ion/app/features/feed/notifications/data/database/notifications_database.m.dart';
+import 'package:ion/app/features/optimistic_ui/database/optimistic_ui_database.m.dart';
+import 'package:ion/app/features/user_block/model/database/block_user_database.m.dart';
+import 'package:ion/app/features/wallets/data/database/wallets_database.m.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'database_manager_service.r.g.dart';
+
+@Riverpod(keepAlive: true)
+DatabaseManagerService databaseManagerService(Ref ref) {
+  final service = DatabaseManagerService();
+  ref.onDispose(service.dispose);
+  return service;
+}
+
+class DatabaseManagerService {
```

Comment:
This manager breaks the open closed principle - any time we need to remove / add a new database, we need to modify this class. The main purpose of this manager is to be able to track when all the databases are closed? If so, can we probably rethink the approach and create some kind of `DatabaseRegister` that will have methods like `.register(db), .remove(db)`, state is Stream<List<db>> - so list of registered databases. So each db will register (on creation) and remove (on logout / user switch) itself from there and we could listen to the state when needed?